pkgs/applications/emulators/dosbox-x/default.nix

```diff
@@ -0,0 +1,87 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, alsa-lib
+, autoconf
+, automake
+, file
+, fluidsynth
+, freetype
+, glib
+, libpcap
+, libpng
+, libslirp
+, libXext
+, libxkbfile
+, libXrandr
+, nasm
+, ncurses
+, pkg-config
+, SDL2
+, which
+, yad
+, zlib
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dosbox-x";
+  version = "2023.09.01";
+
+  src = fetchFromGitHub {
+    owner = "joncampbell123";
+    repo = "dosbox-x";
+    rev = "dosbox-x-v${finalAttrs.version}";
+    hash = "sha256-gEqIPXrq16p6VC9aGW2sRSWE0SG9hiMWeRzSoDvUbB0=";
+  };
+
+  nativeBuildInputs = [
+    autoconf
+    automake
+    file
+    pkg-config
+  ];
+
+  buildInputs = [
+    alsa-lib
+    fluidsynth
+    freetype
+    glib
+    libpcap
+    libpng
+    libslirp
+    libXext
+    libxkbfile
+    libXrandr
+    nasm
+    ncurses
+    SDL2
+    which
+    yad
+    zlib
+  ];
+
+  postPatch = ''
+    substituteInPlace vs/sdlnet/ltmain.sh vs/sdlnet/acinclude/libtool.m4 --replace /usr/bin/file ${file}/bin/file
+    patchShebangs .
+  '';
```

Comment:
I think this is moot now, please give it a whirl and re-open if not.

Reply:
Nope, this is still necessary for cross. Try `nix-build -A dosbox-x --argstr crossSystem aarch64-linux`:
```
checking for sdl-config... no
checking for SDL - version >= 1.2.0... no
*** The sdl-config script installed by SDL could not be found
*** If SDL was installed in PREFIX, make sure PREFIX/bin is in
*** your path, or set the SDL_CONFIG environment variable to the
*** full path to sdl-config.
configure: error: *** SDL version 1.2.0 not found!
```

This is because the script uses `pkg-config` to query the compiler flags for dependencies, while the correct name of the binary for cross-compiling is `${stdenv.cc.targetPrefix}pkg-config` (stored in shell `$PKG_CONFIG` variable for convenience).